### PR TITLE
Build Supply Chain for Web 3.0

### PR DESCRIPTION
**Pull request by herou**
### Project Abstract

We aim to build a Supply chain system that will help communities/businesses and the web3 community would have a system implemented in web3.

### For which grant [level](https://github.com/w3f/Grants-Program#level_slider-levels) are you applying?

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.
- [x] I have read and understood the [FAQs](https://github.com/w3f/Grants-Program/blob/master/docs/faq.md), [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/grant_guidelines_per_category.md) and [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/announcement-guidelines.md).
- [x] A BTC, Ethereum (USDT/USDC/DAI) or Polkadot/Kusama (aUSD) address for the payment of the milestones is provided inside the application.
- [x] I have read and acknowledge the [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/T&Cs.md).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process).
- [x] I prefer the discussion of this application to be in a private Element/Matrix channel. My username is: ``@_______``

### How Did You Hear About our grants program?

- [x] Social Media
- [ ] Hackathon
- [x] Personal Recommendation
- [ ] Substrate Builders Program
- [ ] Investor/VC
- [x] Online Search
- [ ] Other: _______

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the grant application. I have a couple of initial questions: 

- Compared to your [previous grant application](https://github.com/w3f/Grants-Program/pull/1094), you significantly increased the price (weekly rate). What is the reason for this? 
- Do you have a specific use case for this SupplyChain module? Are you working together with a company etc.? 
- For example, I see a lot of legal/privacy problems with this setup. How do you plan to address these? 

I’m personally interested in supply chain-focused modules, but the price point seems relatively high for a purely education-focused project.

**Comment by herou** [comment]
> Thanks for the grant application. I have a couple of initial questions:
> 
> * Compared to your [previous grant application](https://github.com/w3f/Grants-Program/pull/1094), you significantly increased the price (weekly rate). What is the reason for this?
> * Do you have a specific use case for this SupplyChain module? Are you working together with a company etc.?
> * For example, I see a lot of legal/privacy problems with this setup. How do you plan to address these?
> 
> I’m personally interested in supply chain-focused modules, but the price point seems relatively high for a purely education-focused project.

Hi @Noc2, thank you for your quick reply.
1- The first grant was built for education purposes. It was my first application and just one of us worked there. This time we will be two people working full-time on this project who have already proven experience to get the job done.  The amount might be higher but it would take more or less 9 months of work. I do not think it is high for two experienced developers who will work full-time on this project. We tried to ask as less as possible! Remember, this amount will be split between two people.
2- I have already had calls with many companies whose main job is the supply chain. Most of them use big trucks to ship car parts/furniture/heavy things to deliver to the main client and they would love to have at least an MVP product that they can start using. We do not have many supply-chain systems on Substrate, so we thought it would be a great opportunity to build something which is valuable for the businesses and the community.
3- The idea is to build an initial MVP that fixes the main supply chain issues. Of course, it might be things to be enhanced/corrected but we have to take into consideration that we are just two devs working on everything, including designing, implementing, testing, talking to the customers, etc. If you have suggestions, you are more than welcome.

We would be pleased if you can give us the opportunity to build something in which we believe.

If you still have questions, we can schedule a call and try to make it work. 
Thanks again.

**Comment by herou** [comment]
@Noc2 Do you have any suggestion/news? Please, let me know.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. I will mark your application as ready for review so that others can take a look at it as well. 

But the costs still seem high to me. Also, your first milestone has an estimated duration of 2-3 months, an FTE of 2, and costs 30k. Your last milestone has the same estimate but cost 18k, which seems more aligned with your previous grant. Could you also comment on my previous question regarding legal or privacy issues with on-chain supply chain management?

**Comment by herou** [comment]
> Thanks for the update. I will mark your application as ready for review so that others can take a look at it as well.
> 
> But the costs still seem high to me. Also, your first milestone has an estimated duration of 2-3 months, an FTE of 2, and costs 30k. Your last milestone has the same estimate but cost 18k, which seems more aligned with your previous grant. Could you also comment on my previous question regarding legal or privacy issues with on-chain supply chain management?

Well, if you see the amount of work (number of functions) is not equal in all the milestones. We divided the milestones into entities like a manager, client, and courier so would be easier for developing and also for you to test it.
About the legal or privacy issue: We have found a simple solution because we do have a depart that can take care of legal stuff. Every client would have its own private node. We will help them with configuration/setup and everything they need so we are not going to store their data on our chain. In the future client has to use our software as a service. So we are gonna hire a  legal team who can prepare the GDPR contract and all the needed things.

**Comment by herou** [comment]
@takahser @alxs @keeganquigley Can you please review my application? Thank you!

File: applications/sub_supply_chain.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
+            # W3F Grant Proposal
+- **Project Name:** SubSupplyChain
```
Comment (alxs) [inline applications/sub_supply_chain.md:2]: See the current [template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md)

```suggestion
# SubSupplyChain

```

**Comment by alxs** [comment]
@herou thank you for your application. I have to say I agree with David here that the costs seem very high, not so much in terms of rates as with respect to the deliverables. The milestones consist of business logic which seems fairly straightforward to implement, and you provide almost no technical details.

Besides, if each client runs a private node I don't see how this is any different to a traditional database, or what advantages it would bring.

Also, are you aware of https://github.com/OriginTrail?

**Comment by herou** [comment]
> @herou thank you for your application. I have to say I agree with David here that the costs seem very high, not so much in terms of rates as with respect to the deliverables. The milestones consist of business logic which seems fairly straightforward to implement, and you provide almost no technical details.
> 
> Besides, if each client runs a private node I don't see how this is any different to a traditional database, or what advantages it would bring.
> 
> Also, are you aware of https://github.com/OriginTrail?

@alxs Thank you for your feedback. I put the needed details in the application and I made some modifications in general.I left the main methods which are needed because I updated also the price. Please let me know how you find it.
I and my partner decided to hire a legal team who can prepare the GDPR contract and all the needed things.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update and the price reduction! What do you mean by “hire a legal team who can prepare the GDPR contract”? This might be a misunderstanding.

**Comment by herou** [comment]
Hi @Noc2, @alxs . Do you have any updates? Thank you!

**Review by Noc2 — CHANGES_REQUESTED** [review]
I don’t think you need to hire a legal team to look into GDPR. If you are working together with a certain company, they will know the legal requirements and if they can leverage your product or not. I personally won’t approve the application, since I have doubts that anyone is ever going to use it. But this is just my opinion and others might have different opinions.

**Comment by alxs** [comment]
I agree with David here and for the reasons I previously mentioned, I won't approve the application either. But maybe others in the committee think differently and otherwise, you're always welcome to apply again at a later stage or with a different project.

**Comment by herou** [comment]
@alxs @Noc2 Thank you for your feedback. It seems this idea was not very interesting. I will come up with a new idea in the future. Thank you!

**Pull request closed without merging**